mlir/include/mlir/Dialect/Linalg/TransformOps/LinalgTransformOps.td

```diff
@@ -41,6 +41,18 @@ def ApplyEraseUnnecessaryInputsPatternsOp : Op<Transform_Dialect,
   let assemblyFormat = "attr-dict";
 }
 
+def ApplyGeneralizeTensorPackUnpackPatternsOp
```

Comment:
Sorry for asking this question while the naming is following the existing style. I'm confused about the difference between Generalization and Decomposition. Should we unify the terminology? How do you interpret them?

(I could be the root cause who invented the terms when I wrote the patterns. And I feel that they could be renamed to decomposition now.)

Reply:
> I'm confused about the difference between Generalization and Decomposition.

Same and I am glad that you've raised this :) "Decompose" would make more sense to me - "Generalization" tends to be use in the context of Linalg named Ops (e.g. `linalg.matmul` -> `linalg.generic`). I am happy to rename these as "Decomposition" 👍🏻

Reply:
Oh, that's probably why it put me to use generalization in the first place -- because they were prototyped in linalg_ext, and I could be thinking that linalg_ext -> linalg is a generalization. Anyway, I'm +1 on renaming it to decomposition!

Reply:
Renamed in the 2nd commit ... that should probably be a separate PR. WDYT?

Reply:
https://github.com/llvm/llvm-project/pull/116439

Reply:
Can you revert the last two commits in this case?